landscape/client/manager/snapmanager.py

```diff
@@ -0,0 +1,261 @@
+import logging
+from collections import deque
+
+from twisted.internet import task
+
+from landscape.client.manager.plugin import FAILED
+from landscape.client.manager.plugin import ManagerPlugin
+from landscape.client.manager.plugin import SUCCEEDED
+from landscape.client.monitor.snapmonitor import get_installed_snaps
+from landscape.client.snap.http import INCOMPLETE_STATUSES
+from landscape.client.snap.http import SnapdHttpException
+from landscape.client.snap.http import SnapHttp
+from landscape.client.snap.http import SUCCESS_STATUSES
+
+
+class SnapManager(ManagerPlugin):
+    """
+    Plugin that updates the state of snaps on this machine, installing,
+    removing, refreshing, enabling, and disabling them in response to messages.
+
+    Changes trigger SnapMonitor to send an updated state message immediately.
+    """
+
+    def register(self, registry):
```

Comment:
Might be more clear to move some initialization into  an _init_ () like self._snap_http and self.SNAP_METHODS

Reply:
done.